caravel/models.py

```diff
@@ -1664,3 +1665,42 @@ class FavStar(Model):
     class_name = Column(String(50))
     obj_id = Column(Integer)
     dttm = Column(DateTime, default=func.now())
+
+
+class QueryStatus:
+    SCHEDULED = 'SCHEDULED'
+    CANCELLED = 'CANCELLED'
+    IN_PROGRESS = 'IN_PROGRESS'
+    FINISHED = 'FINISHED'
+    TIMED_OUT = 'TIMED_OUT'
+    FAILED = 'FAILED'
+
+
+class Query(Model):
+
+    """ORM model for SQL query"""
+
+    __tablename__ = 'query'
+    id = Column(Integer, primary_key=True)
+
+    database_id = Column(Integer, ForeignKey('dbs.id'), nullable=False)
+    table_names = Column(Integer, ForeignKey('tables.id'), nullable=True)
+    # Add comma seperate table names as not all of them would be in the
+    # tables table.
+
+    # Store the tmp table into the DB only if the user asks for it.
+    tmp_table_name = Column(String(64))
+    user_id = Column(Integer, ForeignKey('ab_user.id'), nullable=True)
+
+    # models.QueryStatus
+    query_status = Column(String(16))
```

Comment:
to me if the table is name query the field name doesn't have to be prefixed unless there could be confusion. `status` in a Query table clearly points to the state of the query.